README.md

```diff
@@ -15,8 +15,80 @@ following entitlements are supported:
+* UA client auto-enables any services described with `obligations:{enableByDefault: true}`
+POST `<contractToken>` to contracts server API @ https://contracts.canonical.com/api/v1/context/machines/token
+* UA client writes the API response to the root-readonly /var/lib/ubuntu-advantage/machine-token.json
+
+### Enabling a service
+Each service controlled by UA client will have a python module in uaclient/entitlements/\*.py which handles setup and teardown of services when enabled or disabled.
+
+If a contract entitles a machine to a service, `root` user can enable the service with `ua enable <service>`.  If a service can be disabled `ua disabled <service>` will be permitted.
+
+The goal of the UA client is to remain simple and flexible and let the
+contracts backend drive dynamic changes in contract offerings and contraints.
+In pursuit of that goal, the UA client obtains most of it's service contraints
+from a machine token that it ob
+
+The UA Client is simple in that it relies on the machine token on the attached
+machine to describe whether a service is applicable for an envronment and what
+configuration is required to properly enable that service.
+
+Any interactions with the Contract server API are defined as UAContractClient class methods in uaclient/contract.py.
+
+## Directory layout
+The following describes the intent of UA client related directories:
+
+
+| File/Directory | Intent |
+| -------- | -------- |
+| ./tools | Helpful scripts used to publish, release or test various aspects of  UA client |
+| .features/ | Behave BDD integration tests for UA Client
+| uaclient/ | collection of python modules which will be packaged into ubuntu-advantage-tools package to deliver the UA Client CLI |
+| uaclient.entitlements | Service-specific \*Entitlement class definitions which perform enable, disable, status, and entitlement operations etc. All classes derive from base.py:UAEntitlement and many derive from repo.py:RepoEntitlement |
+| uaclient.cli | The entry-point for the command-line client
+| uaclient.clouds | Cloud-platform detection logic used in Ubuntu Pro to determine if a given should be auto-attached to a contract |
+| uaclient.contract | Module for interacting with the Contract Server API |
+| ./demo | Various stale developer scripts for setting up one-off demo environemnts. (Not needed often)
+| /etc/ubuntu-advantage/uaclient.conf | Configuration file for the UA client.|
+| apt-hook | the C++ apt-hook delivering MOTD and apt command notifications about UA support services |
+| apt-conf.d | apt config files delivered to /etc/apt/apt-conf.d to automatically allow unattended upgrades of ESM  security-related components |
+| /var/lib/ubuntu-advantage/private | `root` read-only directory containing Contract API responses, machine-tokens and service credentials |
+ | /var/log/ubuntu-advantage.log | `root` read-only log of ubuntu-advantage operations |
+
```

Comment:
For this section in general, just be consistent with the `./`
E.g., .`./tools`, `.features/`, and `uaclient/` are all different even though they all refer to directories.

Reply:
thanks, did this and moved all the non-relative file paths mentioned to the bottom.